modules/social_features/social_event/social_event.module

```diff
@@ -282,28 +284,108 @@ function _social_event_format_date($event, $view_mode) {
   $time_format = 'social_time';
 
   if (!empty($start_datetime)) {
+    $start_date = $date_formatter->format($start_datetime, $date_format);
+    // Default time should not be displayed.
+    $start_time = $date_formatter->format($start_datetime, $time_format, '', 'UTC') === '00:01' ? '' : $date_formatter->format($start_datetime, $time_format);
+
+    if (!empty($end_datetime)) {
+      $end_date = $date_formatter->format($end_datetime, $date_format);
+      // Default time should not be displayed.
+      $end_time = $date_formatter->format($end_datetime, $time_format, '', 'UTC') === '00:01' ? '' : $date_formatter->format($end_datetime, $time_format);
+    }
+
     // Date are the same or there are no end date.
     if (empty($end_datetime) || $start_datetime == $end_datetime) {
-      $date = $date_formatter->format($start_datetime, $date_format);
-      $time = $date_formatter->format($start_datetime, $time_format);
-      $event_date = "$date - $time";
+      $event_date = empty($start_time) ? $start_date : "$start_date - $start_time";
```

Comment:
If we want to be consistent with the other cases where a dash (`-`) is used to denote "from - to" then we should remove the dash here.